G_channels.md

```diff
@@ -249,6 +249,24 @@ socket = assign(socket, :user, msg["user"])
 
 Sockets store assigned values as a map in `socket.assigns`.
 
+#### Failure Semantics
+
+Servers will restart & clients will fail - understanding how Phoenix responds to these events and the guarantees it offers will  help you design robust systems. 
+
+- Subscriptions
+ 
+On the server, subscriptions are held in an ETS table (which resides in memory). If a server restarts or crashes, the subscriptions will need to be recreated for clients to receive updaes. Fortunately, this functionality is built into the Phoenix javascript client. The clients will attempt to reconnect to the server. Once they do, they'll rejoin the channels they were subscribed to using exponential back off.
+
+- Channel Crashes
+
+If a channel crashes on the server, the client is notified and the `channel.onError` callback is invoked. The client then rejoins using exponential back off.
+
+- Messages
+
+On the client, messages are queued into a `PushBuffer` and sent to the server when there is a connection. If there isn't a connection available, they'll be held in memory until a connection is established.  The messages are not persisted in LocalStorage or IndexedDb, so if the client's browser tab closes before the messages are sent, the messages are lost.
```

Comment:
"they'll be held in memory until a connection is established, or until they receive a "timeout" event, which defaults to 5s."